src/createReferenceObject.js

```diff
@@ -0,0 +1,27 @@
+const createBoundingBox = require('./createBoundingBox');
+
+//Create a reference object for an element
+module.exports.getRef = function (target, userOptions) {
+    //Define popper reference object and cy reference  object
+    let refObject = {
+
+        //Store copies of dimensions and cyElement objects
+        cyElement: target,
+
+        //Define the bounding box for the popper target
+        getBoundingClientRect: userOptions.boundingBox ? userOptions.boundingBox : function () {
```

Comment:
Is there a case where we need `options.boundingBox`?  Or is it enough to use `options.position()` and `options.getDimensions()` to build the rect?

Reply:
`Options.boundingBox` allows the user to pass in a pre formatted bounding box into popper. The initial idea was that without the option to override the bounding box, if the user had a bounding box from elsewhere (e.g `.popperRef()), then they would have to manually extract `position()` and `getDimensions()`

Essentially it was added so you could just skip the process of having to define the `position()` function if you already had a bounding box from elsewhere.